src/tools.js

```diff
@@ -220,13 +220,33 @@ tools = [{
 	description: "Fills an area with the selected drawing color.",
 	cursor: ["fill-bucket", [8, 22], "crosshair"],
 	pointerdown: function(ctx, x, y){
-		
 		// Get the rgba values of the selected fill color
 		var rgba = get_rgba_from_color(fill_color);
-		
-		// Perform the fill operation
-		draw_fill(ctx, x, y, rgba[0], rgba[1], rgba[2], rgba[3]);
-		
+
+		$G.one("pointerdown", function(e) {
```

Comment:
This isn't an ideal way to do this (access the shift key state), but I guess there isn't really an obvious one. I didn't think about that. But the event could be exposed to the `pointerdown: function(ctx, x, y){ ... }` function (or maybe the whole system should be reworked and made less "frameworky", and just have a hook for setting up event listeners normally like this, instead of the  `pointerdown: function(ctx, x, y){ ... }`)